crates/ethrpc/src/alloy/errors.rs

```diff
@@ -0,0 +1,90 @@
+use alloy::{contract::Error as ContractError, transports::RpcError};
+
+/// Bubbles up node errors, ignoring all other errors.
+pub fn ignore_non_node_error<T>(result: Result<T, ContractError>) -> anyhow::Result<Option<T>> {
+    match result {
+        Ok(result) => Ok(Some(result)),
+        Err(err) if err.is_node_error() => Err(err.into()),
+        Err(_) => Ok(None),
+    }
+}
+
+pub trait ContractErrorExt {
+    /// Returns whether a given error is a contract error, this is considered to
+    /// be all errors except the transport error.
```

Comment:
The comment mentions transport errors but the other function is called `is_node_error`